packages/contracts/contracts/ArtonomousArtPieceToken.sol

```diff
@@ -1,14 +1,62 @@
 pragma solidity ^0.4.24;
 
 import "openzeppelin-solidity/contracts/token/ERC721/ERC721Token.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "./Metadata.sol";
 
-contract ArtonomousArtPieceToken is ERC721Token {
+contract ArtonomousArtPieceToken is ERC721Token, Ownable {
+    address metadata;
+    mapping(uint => string) generators;
 
-    constructor(string name, string symbol) public ERC721Token(name, symbol) {
+    constructor(string name, string symbol) public ERC721Token(name, symbol) {}
+
+    /**
+    * @dev For returning the URL of the JSON object describing the token.
+    * @return A string of the URL.
+    */
+    function getGenerator(uint _block) public view returns (string generator) {
+        return generators[_block];
```

Comment:
we could just make the mapping public, right?

Reply:
right you are